File: core/src/main/java/com/neqsoft/reso/cpu/CpuInfoFragment.java
package com.neqsoft.reso.cpu;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.neqsoft.reso.R;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import static androidx.lifecycle.ViewModelProviders.of;

public class CpuInfoFragment extends Fragment {

  @Nullable
  private CpuInfoRecyclerAdapter cpuInfoRecyclerAdapter;
  @Nullable
  private CpuInfoViewModel cpuInfoViewModel;

  @Override
  public void onAttach(@NonNull Context context) {
    super.onAttach(context);
  }

  @Override
  public void onCreate(@Nullable Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    cpuInfoViewModel = of(this).get(CpuInfoViewModel.class);
  }

  @Nullable
  @Override
  public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
    View view = inflater.inflate(R.layout.fragment_cpu_info, container, false);
    bind(view);
    if (cpuInfoViewModel != null)
      cpuInfoViewModel.getCpuInfo().observe(this, this::display);
    return view;
  }

  private void bind(final View view) {
    RecyclerView recyclerView = view.findViewById(R.id.cpuInfoRv);
    recyclerView.setLayoutManager(new LinearLayoutManager(getContext()));
    recyclerView.setAdapter(cpuInfoRecyclerAdapter);
  }

  private void display(@Nullable CpuInfo cpuInfo) {
    cpuInfoRecyclerAdapter.submit(cpuInfo);
  }
}

File: core/src/main/java/com/neqsoft/reso/cpu/CpuInfoFragment.kt
package com.neqsoft.reso.cpu

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.neqsoft.reso.R

class CpuInfoFragment : Fragment() {
    private val cpuInfoRecyclerAdapter: CpuInfoRecyclerAdapter? = null
    private var cpuInfoViewModel: CpuInfoViewModel? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        cpuInfoViewModel = ViewModelProvider(this).get(CpuInfoViewModel::class.java)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_cpu_info, container, false)
        bind(view)
        return view
    }

    private fun bind(view: View) {
        val recyclerView: RecyclerView = view.findViewById(R.id.cpuInfoRv)
        recyclerView.layoutManager = LinearLayoutManager(context)
        recyclerView.adapter = cpuInfoRecyclerAdapter
    }

    private fun display(cpuInfo: CpuInfo?) {
        cpuInfoRecyclerAdapter?.submit(cpuInfo)
    }
}